### ISSUE TITLE
Misleading value duplication / rounding when using SI

### ISSUE THREAD

**Issue by e-foley** [OpenRCT2/OpenRCT2#2519]
#2512 added SI units as a system of measurement.  I love SI units and they generally work well across the game, but there are some subtle effects that are a little weird (v0.0.4 build 1552).

Because 1 m/s > 1 mph, speed values get "duplicated" for those values that change by 1 mph at a time.  For example, build a side-friction roller coaster and change the lift hill speed.  In mph, the value changes from 3 mph to 4 mph, but using SI units, the number sticks at 1 m/s, *appearing* not to change.  Of course, internally, the speed is changing by a whole 33%, but to the player, their modification has no observable effect.  Chairlift speeds are especially confusing since they start at 0 m/s, implying the ride is stopped by default.

In short, the addition of SI units makes the effect of certain UI elements ambiguous.  Regardless of whether there's a simple fix for this, I find this behavior worth reporting.

The nicest way to address this might be to calculate 10 times the current value and stick a decimal point between the rightmost two numbers, forming values like 1.7 m/s or 10.0 m/s.  Currency formats can handle moving decimal places ($ has them where ¥ doesn't), so this gives me hope.

**Comment by janisozaur** [comment]
The game itself computes values in mph, and the same issue as you reported
happens when you choose kmph. I'm not sure if this is something we can fix
already.

**Comment by HaasJona** [comment]
Well it is somewhat fixed but now the vehicle speed jumps between discrete values. The mph value we use as input should probably have higher precision in the long run. Currently it's rounded to integers.

**Comment by IntelOrca** [comment]
@HaasJona Indeed, it has to convert the speed into mph from the in-game units:
```mph = (abs(vehicle->velocity) * 9) >> 18```

But this is done everywhere that draws a speed, so each one of those places would have to be changed to use some universal function that goes straight from vehicle speed to target units.
